controllers/imagecollector/imagecollector_controller.go

```diff
@@ -0,0 +1,324 @@
+			return reconcile.Result{}, fmt.Errorf("Could not delete failed imagejob")
+		}
+	default:
+		log.Info("should not reach this point for imagejob", relevantJobs[0])
+	}
+
+	return ctrl.Result{RequeueAfter: time.Second * 30}, nil
+}
+
+func isNotFound(err error) bool {
+	if err != nil && client.IgnoreNotFound(err) == nil {
+		return true
+	}
+	return false
+}
+
+func (r *Reconciler) handleJobDeletion(ctx context.Context, job *eraserv1alpha1.ImageJob) (ctrl.Result, error) {
+	if job.Status.DeleteAfter == nil {
+		return ctrl.Result{}, nil
+	}
+
+	until := time.Until(job.Status.DeleteAfter.Time)
+	if until > 0 {
+		log.Info("Delaying imagejob delete", "job", job.Name, "deleteAter", job.Status.DeleteAfter)
+		return ctrl.Result{RequeueAfter: until}, nil
+	}
+
+	log.Info("Deleting imagejob", "job", job.Name)
+	err := r.Delete(ctx, job)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) createImageJob(ctx context.Context, req ctrl.Request, imageCollector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	job := &eraserv1alpha1.ImageJob{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "imagejob-",
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(imageCollector, schema.GroupVersionKind{Group: "eraser.sh", Version: "v1alpha1", Kind: "ImageCollector"}),
+			},
+		},
+		Spec: eraserv1alpha1.ImageJobSpec{
+			JobTemplate: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:            "collector",
+							Image:           *collectorImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"cpu":    resource.MustParse("7m"),
+									"memory": resource.MustParse("25Mi"),
+								},
+								Limits: corev1.ResourceList{
+									"cpu":    resource.MustParse("8m"),
+									"memory": resource.MustParse("30Mi"),
+								},
+							},
+						},
+					},
+					ServiceAccountName: "eraser-controller-manager",
+				},
+			},
+		},
+	}
+
+	err := r.Create(ctx, job)
+	if err != nil {
+		log.Info("Could not create collector ImageJob")
+		return reconcile.Result{}, err
+	}
+
+	log.Info("Successfully created collector ImageJob", "job", job.Name)
+	return reconcile.Result{}, nil
+}
+
+func (r *Reconciler) updateSharedCRD(ctx context.Context, req ctrl.Request, imageCollector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	imageCollectorList := &eraserv1alpha1.ImageCollectorList{}
+	if err := r.List(ctx, imageCollectorList); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	items := imageCollectorList.Items
+
+	var list []eraserv1alpha1.Image
+
+	for i := range items {
+		temp := items[i].Spec.Images
+		for _, img := range temp {
+			list = append(list, eraserv1alpha1.Image{Name: img.Name, Digest: img.Digest})
```

Comment:
Be sure to skip the shared ImageCollector:
```
if img.Name == "imagecollector-shared" {
    continue
}
```

"imagecollector-shared" should be actually defined as a constant, though.

Reply:
I think there wouldn't be an imagecollector-shared image the nodes

Reply:
?

Reply:
I'm not sure I follow

Reply:
got confused since it said img.Name, but added a skip for the collector with items[i].Name